cmake/modules/hunter_check_toolchain_for_define.cmake

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2016 NeroBurner
+# All rights reserved.
+
+include(hunter_test_string_not_empty)
+
+function(hunter_check_toolchain_for_define definition output_var)
+  # the toolchain path must be set
+  hunter_test_string_not_empty("${HUNTER_TOOLCHAIN_ID_PATH}")
+  set(TOOLCHAIN_INFO_FILE "${HUNTER_TOOLCHAIN_ID_PATH}/toolchain.info")
+
+  # read toolchain.info file into memory, each line a entry in a list
+  file(STRINGS "${TOOLCHAIN_INFO_FILE}" toolchain_contents)
+
+  # clean ouptut variable
+  set(definition_found)
+
+  foreach(line ${toolchain_contents})
+    # find the specified define
+    string(REGEX MATCH "^#define ${definition} .*$" match "${line}")
+    if(match)
+      # extract the definition
+      string(REGEX REPLACE "^#define ${definition} " "" definition_found "${match}")
```

Comment:
> renamed the variable

I didn't said that the name of variable is wrong :) So now if we have toolchain.info:
```
#define A
```
and what to check A (defined) and B (not defined), we will have the same results:
```cmake
hunter_check_toolchain_for_define("A" a_defined)
hunter_check_toolchain_for_define("B" b_defined)
if(a_defined STREQUAL "")
  message("A not defined")
endif()
if(b_defined STREQUAL "")
  message("B not defined")
endif()
```
I guess output will be:
```
A not defined
B not defined
```
which is wrong, or at least confusing.

There are two traits of macro in C++. One is about is the macro defined or not, and the second is about the value of macro. So if you're not sure you should write `#if defined(SOMETHING) && (SOMETHING > 3)`. Same is here, I was thinking that this function should work similar to `#if defined(SOMETHING)`, looks like you're trying to get the value.